### DISCUSSION TITLE
how to do conditional find unique for username/email authentication

### DISCUSSION THREAD

**Discussion by moharifhidayat7** [prisma/prisma#12345] [Q&A] [answered]
I am trying to make username/email authentication. so user can login with either username or email.
is it possible to do it in one query?

this is what I tried
```
const user = await prisma.user.findUnique({
    where: {
        OR: [{
                username: credentials.username,
            },
            {
                email: credentials.email,
            },
        ],
    },
});
```

**Comment by nurul3101** [comment, marked as answer]
Hey @moharifhidayat7 👋 

This query should work, are you facing any unexpected behaviour?

Ideally, In your API you would know that if username is received or email is received and based on that you can make appropriate call with email or username.

**Reply by vladi-stevanovic** [reply]
👋 Hello @moharifhidayat7  I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! 🙌

**Reply by brandensilva** [reply]
It seems like the OR clause isn't supported in FindUnique. It does appear to be supported in FindFirst.

Here is the typescript error I see when I add OR for searching by username or email:

``Type '{ OR: ({ username: string | undefined; } | { email: string | undefined; })[]; }' is not assignable to type 'UserWhereUniqueInput'.
  Object literal may only specify known properties, and 'OR' does not exist in type 'UserWhereUniqueInput'``

**Reply by nurul3101** [reply]
@brandensilva You can use OR in findUnique clause once you have enabled [extendedWhereUnique](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#filter-on-non-unique-fields-with-userwhereuniqueinput) preview feature.
<img width="467" alt="Screenshot 2022-12-14 at 6 53 52 PM" src="https://user-images.githubusercontent.com/13049676/207606833-51aefab9-cea8-4f62-9eea-7efd25d64bd7.png">

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
